### PR TITLE
refactor: Rename msgtype ERROR to FAILURE

### PR DESCRIPTION
**Pull request by w4bremer**

### CODE REVIEW

File: src/olink/core/types.cpp [hunk 1]
```diff
@@ -114,7 +114,7 @@ std::string toString(MsgType type) {
         { MsgType::INVOKE, "invole" },
```
Comment (jryannel) [inline src/olink/core/types.cpp:114]: typo: invole => invoke

File: src/olink/core/types.cpp [hunk 2]
```diff
@@ -114,7 +114,7 @@ std::string toString(MsgType type) {
...
         { MsgType::INVOKE_REPLY, "invoke_reply" },
         { MsgType::SIGNAL, "signal" },
-        { MsgType::ERROR, "error"
```
Comment (jryannel) [inline src/olink/core/types.cpp:117]: should error not be failure?

File: src/olink/core/protocol.cpp [hunk 3]
```diff
@@ -113,7 +113,7 @@ json Protocol::signalMessage(std::string name, json args)
 json Protocol::errorMessage(MsgType msgType, int requestId, std::string error)
```
Comment (jryannel) [inline src/olink/core/protocol.cpp:113]: should be `failureMessage`

File: tests/test_protocol.cpp [hunk 4]
```diff
@@ -81,7 +81,7 @@ TEST_CASE("protocol")
     }
     SECTION("error") {
         json msg = Protocol::errorMessage(msgType, requestId, error);
```
Comment (jryannel) [inline tests/test_protocol.cpp:83]: failureMessage

**Review by jryannel — CHANGES_REQUESTED** [review]

**Comment by w4bremer** [comment]
New change is at https://github.com/apigear-io/objectlink-core-cpp/pull/new/bugfix/renameMgstypeToCapitalCase

**Pull request closed without merging**